pkg/cmd/cli/cmd/new/newapp.go

```diff
@@ -0,0 +1,383 @@
+package new
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	"github.com/openshift/origin/pkg/cmd/cli/cmd"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	dockerutil "github.com/openshift/origin/pkg/cmd/util/docker"
+	"github.com/openshift/origin/pkg/dockerregistry"
+	"github.com/openshift/origin/pkg/generate/app"
+	"github.com/openshift/origin/pkg/generate/dockerfile"
+	"github.com/openshift/origin/pkg/generate/source"
+)
+
+type newAppConfig struct {
+	SourceRepositories util.StringList
+
+	Components   util.StringList
+	ImageStreams util.StringList
+	DockerImages util.StringList
+	Groups       util.StringList
+	Environment  util.StringList
+
+	TypeOfBuild string
+
+	localDockerResolver Resolver
+	imageStreamResolver Resolver
+
+	searcher Searcher
+	detector Detector
+}
+
+const longNewAppDescription = `
+Create a new application in OpenShift by specifying source code, templates, and/or images.
+
+Examples:
+  $ osc new-app .
+  <try to create an application based on the source code in the current directory>
+
+  $ osc new-app mysql
+  <use the public DockerHub MySQL image to create an app>
+
+  $ osc new-app myregistry.com/mycompany/mysql
+  <use a MySQL image in a private registry to create an app>
+
+  $ osc new-app openshift/ruby-20-centos~git@github.com/mfojtik/sinatra-app-example
+  <build an application using the OpenShift Ruby DockerHub image and an example repo>`
+
+func NewCmdNewApplication(f *cmd.Factory, out io.Writer) *cobra.Command {
+	config := newAppConfig{
+		searcher: &mockSearcher{},
+		detector: sourceRepositoryEnumerator{
+			detectors: source.DefaultDetectors,
+			tester:    dockerfile.NewTester(),
+		},
+	}
+	helper := dockerutil.NewHelper()
+
+	cmd := &cobra.Command{
+		Use:   "new-app <components> [--code=<path|url>]",
+		Short: "Create a new application",
+		Long:  longNewAppDescription,
+
+		Run: func(c *cobra.Command, args []string) {
+			if dockerclient, _, err := helper.GetClient(); err == nil {
+				config.localDockerResolver = dockerClientResolver{dockerclient}
+				config.localDockerResolver = dockerRegistryResolver{dockerregistry.NewClient()}
```

Comment:
I imagine that we want a separate resolver field for the docker registry on the newAppConfig